smoke-tests/apps/SamplingOverrides/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/SamplingOverrides3Test.java

```diff
@@ -0,0 +1,109 @@
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_11_OPENJ9;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_17;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_17_OPENJ9;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_21;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_21_OPENJ9;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_23;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_23_OPENJ9;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_8;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_8_OPENJ9;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.WILDFLY_13_JAVA_8;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.WILDFLY_13_JAVA_8_OPENJ9;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@UseAgent("applicationinsights3.json")
+abstract class SamplingOverrides3Test {
+
+  @RegisterExtension static final SmokeTestExtension testing = SmokeTestExtension.create();
+
+  @Test
+  @TargetUri(value = "/health-check", callCount = 100)
+  void testSampling() throws Exception {
+    // super super low chance that number of sampled requests is less than 25
+    long start = System.nanoTime();
+    while (testing.mockedIngestion.getCountForType("RequestData") < 25
+        && NANOSECONDS.toSeconds(System.nanoTime() - start) < 10) {}
+    // wait ten more seconds to before checking that we didn't receive too many
+    Thread.sleep(SECONDS.toMillis(10));
+    int requestCount = testing.mockedIngestion.getCountForType("RequestData");
+    int dependencyCount = testing.mockedIngestion.getCountForType("RemoteDependencyData");
+    int logCount = testing.mockedIngestion.getCountForType("MessageData");
+    // super super low chance that number of sampled requests/dependencies/traces
+    // is less than 2 or greater than 20
+    assertThat(requestCount).isGreaterThanOrEqualTo(2);
+    assertThat(requestCount).isLessThanOrEqualTo(20);
+    assertThat(dependencyCount).isGreaterThanOrEqualTo(2);
+    assertThat(dependencyCount).isLessThanOrEqualTo(20);
+    assertThat(logCount).isGreaterThanOrEqualTo(2);
+    assertThat(logCount).isLessThanOrEqualTo(20);
+
+    testing
+        .mockedIngestion
+        .getItemsEnvelopeDataType("RequestData")
+        .forEach(
+            item -> {
+              assertThat(item.getSampleRate()).isEqualTo(10);
+            });
+    testing
```

Comment:
Perhaps add a comment that in this case the sampling percentage should be that of the request.